src/nncf/quantization/algorithms/weight_compression/algorithm.py

```diff
@@ -345,27 +349,37 @@ def set_backend_entity(self, model: TModel) -> None:
 
         :param model: Backend-specific input model.
         """
-        model_backend = get_backend(model)
-        if model_backend == BackendType.OPENVINO:
+        self._model_backend = get_backend(model)
+        if self._model_backend == BackendType.OPENVINO:
             from nncf.quantization.algorithms.weight_compression.openvino_backend import OVWeightCompressionAlgoBackend
 
             self._backend_entity = OVWeightCompressionAlgoBackend(model)
-        elif model_backend == BackendType.TORCH:
+        elif self._model_backend == BackendType.TORCH:
             from nncf.quantization.algorithms.weight_compression.torch_backend import PTWeightCompressionAlgoBackend
 
             self._backend_entity = PTWeightCompressionAlgoBackend()
-        elif model_backend == BackendType.TORCH_FX:
+        elif self._model_backend == BackendType.TORCH_FX:
             from nncf.quantization.algorithms.weight_compression.torch_fx_backend import FXWeightCompressionAlgoBackend
 
             self._backend_entity = FXWeightCompressionAlgoBackend()
-        elif model_backend == BackendType.ONNX:
+        elif self._model_backend == BackendType.ONNX:
             from nncf.quantization.algorithms.weight_compression.onnx_backend import ONNXWeightCompressionAlgoBackend
 
             self._backend_entity = ONNXWeightCompressionAlgoBackend(model)
         else:
-            msg = f"Cannot return backend-specific entity because {model_backend.value} is not supported!"
+            msg = f"Cannot return backend-specific entity because {self._model_backend.value} is not supported!"
             raise nncf.UnsupportedBackendError(msg)
 
+    def get_ignored_patterns(self) -> GraphPattern:
+        """
+        Returns backend-specific ignored patterns for the Weight Quantization algorithm.
+
+        :return: Backend-specific ignored patterns for the Weight Quantization algorithm
+        """
+        patterns = Patterns()
+        patterns.register(self._backend_entity.create_rope_pattern(), "rope")
```

Comment:
Add one backend specific function that will return list of patterns, to add patterns in them without adding new functions for each patterns.

Reply:
Then we will have different set of patterns for different backends, do you think this way is sustainable?
 I did exactly as it was suggested by you in the last comment

Reply:
I did not suggest to add function create_rope_pattern.
PatternManager is not guaranty that set of patterns is equal too